docs/reference/catalog.rst

```diff
@@ -0,0 +1,96 @@
+=========
+Catalog
+=========
+.. currentmodule:: lsdb.catalog
+
+Catalog Types
+~~~~~~~~~~~~~~~~~~
+.. autosummary::
+    :toctree: api/
+
+    Catalog
+    MarginCatalog
+    MapCatalog
+
+Properties
+~~~~~~~~~~~~~~~~~~
+.. autosummary::
+    :toctree: api/
+
+    Catalog.hc_structure
+    Catalog.dtypes
+    Catalog.columns
+    Catalog.nested_columns
+    Catalog.all_columns
```

Comment:
Welcome, much-used property!  Will this mean that I can refer to it as
```reStructuredText
:attr:`Catalog.all_columns`
```
? That would be lovely!

Reply:
I think you might need to use the full(-er) path of `lsdb.catalog.Catalog.all_columns`, since the `lsdb.catalog` bit is in the `currentmodule` attribute above for convenience.